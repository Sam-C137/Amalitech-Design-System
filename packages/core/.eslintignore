dist
loader
www
**/*.e2e.ts
**/*.spec.tsx
**/*.spec.ts
