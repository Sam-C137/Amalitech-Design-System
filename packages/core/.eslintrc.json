{
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "extends": [
    "plugin:@stencil-community/recommended"
  ],
  "ignorePatterns": [
    "**/*.stories.tsx",
    "**/*.spec.*",
    "**/*.e2e.*"
  ],
  "rules": {
    "react/jsx-no-bind": "off"
  }
}
