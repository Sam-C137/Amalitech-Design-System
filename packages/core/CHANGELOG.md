# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.8.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.2...v0.8.3) (2024-07-18)


### Bug Fixes

* min max value attribute changes ([8276d9d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/8276d9dd0d176071116497f01c12c92caecd5706))





## [0.8.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.1...v0.8.2) (2024-06-02)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.8.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.0...v0.8.1) (2024-06-01)


### Bug Fixes

* tests ([05989c4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/05989c4eaa674a902264d87c3044b6eb91794698))





# [0.8.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.7.0...v0.8.0) (2024-05-27)


### Features

* add double border to radio ([21d478d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/21d478d51df6c38aa88b8a1b429b76baa7371040))





# [0.7.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.2...v0.7.0) (2024-05-14)


### Features

* add clear method to select ([689be3d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/689be3df7d4c3f895ef8b2f11fc8bc24a2f78ab3))
* create select value accessor directive ([d5957c8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/d5957c85d858f02ec8506671ae7ac142421a34c5))





## [0.6.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.1...v0.6.2) (2024-05-06)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.6.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.0...v0.6.1) (2024-05-06)


### Bug Fixes

* removed default switch margin ([503f85d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/503f85da2f380b6b7a89e3a46015ea367d67a2ce))





# [0.6.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.5.0...v0.6.0) (2024-04-30)


### Bug Fixes

* pagination spacing ([00c99e4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/00c99e4e17edf1f4d145135d8f1da444f6cb6ed6))


### Features

* exposed totalPagesCount in pagination ([4105737](https://github.com/Amali-Tech/Amalitech-Design-System/commit/4105737d72ba70a4bc0b9a7fa67dca5e73c2bc5b))





# [0.5.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.4.0...v0.5.0) (2024-04-28)


### Features

* utilities align self ([ab106b6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/ab106b6f3752e20155c9b307d89cbcc4b12c79ec))





# [0.4.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.7...v0.4.0) (2024-04-28)


### Features

* add z-index to utilities ([75efd78](https://github.com/Amali-Tech/Amalitech-Design-System/commit/75efd78fbf0a22ce34139c7569a5ee894faaa088))





## [0.3.7](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.6...v0.3.7) (2024-04-27)


### Bug Fixes

* date picker open method: ([710e97b](https://github.com/Amali-Tech/Amalitech-Design-System/commit/710e97b29a3538231ada47ffddd7314be1766559))





## [0.3.6](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.5...v0.3.6) (2024-04-24)


### Bug Fixes

* centered the checkbox in it's label ([e7fbbaa](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e7fbbaaf3f79a6d0185777a0ce52a44a6b6f84e8))






## [0.3.5](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.4...v0.3.5) (2024-04-15)


### Bug Fixes

* build ([12140c9](https://github.com/Amali-Tech/Amalitech-Design-System/commit/12140c95e8be561d9f26d67c15b71c2beb50dee0))
* build ([a38dd2e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a38dd2e615bedc32ab7a68ce841a9b04e4f8267b))
* card tests ([5c99406](https://github.com/Amali-Tech/Amalitech-Design-System/commit/5c99406dc9796839ee06326fdaf2b8471e2d554d))
* starting with a value ([fb24ce3](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fb24ce3cf7510dc5cca9c868aa03b48e3aa8acdb))
* tests ([373434e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/373434e70de883671a1f828d1cbda137d717978a))
* tests ([eb11522](https://github.com/Amali-Tech/Amalitech-Design-System/commit/eb11522d23bf5b31c41228ba9f522998641c677f))
* tests ([e44f72d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e44f72d42a5b9fc587edfbeecafd3c1dc89b799a))
* tests ([799a98c](https://github.com/Amali-Tech/Amalitech-Design-System/commit/799a98cf03b6c008f749fdb754187abcca9cface))
* tests ([fcb7f47](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fcb7f47051af324d7d5ea1c0a285f5f2645788be))
* tests ([66ea8f8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/66ea8f888460610f550d88b51f1d4ec870444072))






## [0.3.4](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.3...v0.3.4) (2024-04-05)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.2...v0.3.3) (2024-04-04)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.1...v0.3.2) (2024-04-03)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.3.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.0...v0.3.1) (2024-04-03)


### Bug Fixes

* tests ([a1ca9e2](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a1ca9e26ce368e9d26b114605ad3869ccaa5dd8d))






# [0.3.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.2...v0.3.0) (2024-04-02)


### Features

* form submission for input & button ([2cc64d6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/2cc64d69378de5a3ec3e7c9da20f53e36e3a5dcc))






## [0.2.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.1...v0.2.2) (2024-03-28)

**Note:** Version bump only for package @amalitech-design-system/core
