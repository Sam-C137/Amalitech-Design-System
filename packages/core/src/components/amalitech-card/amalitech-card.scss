@import '../../global/variables.scss';

/**
* Exposed css variables
*
* @prop --backgroundColor: Background color of the card, default is `#f6f6f8`;
* @prop --textColor: Text color of the card, default is `#474d66`;
* @prop --card-shadow: Box shadow of the card, default is `0px 8px 32px 0px hsla(0, 0%, 0%, 0.1), 0px 4px 8px 0px hsla(0, 0%, 0%, 0.1)`;
* @prop --card-shadow-hover: Box shadow of the card on hover, default is `0px 8px 32px 0px hsla(0, 0%, 0%, 0.2), 0px 4px 8px 0px hsla(0, 0%, 0%, 0.2)`;
* @prop --card-radius: Border radius of the card, default is `0.5rem`;
* @prop --card-body-padding-vertical: Padding top and bottom of the card body, default is `1.25rem`;
* @prop --card-body-padding-horizontal: Padding left and right of the card body, default is `1.25rem`;
*/

$card-radius: 0.5rem !default;

$card-body-padding-vertical: 1.25rem !default;
$card-body-padding-horizontal: 1.25rem !default;

$card-header-padding-vertical: 1.2rem !default;
$card-header-padding-horizontal: 1.2rem !default;
$card-header-shadow: 0 0.125em 0.25em rgba($dark-bg, 0.1) !default;
$card-header-radius-top: 0.5rem !default;
$card-header-radius-bottom: 0 !default;

$card-image-radius-top: 0.25rem !default;
$card-image-radius-bottom: 0.25rem !default;

$card-shadow: 0px 8px 32px 0px hsla(0, 0%, 0%, 0.1), 0px 4px 8px 0px hsla(0, 0%, 0%, 0.1) !default;
$card-shadow-hover: 0px 8px 32px 0px hsla(0, 0%, 0%, 0.2), 0px 4px 8px 0px hsla(0, 0%, 0%, 0.2) !default;

:host {
  --card-background-color: var(--backgroundColor, $light-bg);
  --card-text-color: var(--textColor, $dark-bg);
  display: block;
  position: relative;
  overflow: hidden;
  box-shadow: var(--card-shadow, $card-shadow);
  border-radius: var(--card-radius, $card-radius);
  transition: all 0.3s;
}

:host(.with-hover-animation:hover) {
  transition: all 0.3s;
  box-shadow: var(--card-shadow-hover, $card-shadow-hover);
  cursor: pointer;
}

:host a {
  text-decoration: none;
  cursor: pointer;
  user-select: none;
  -webkit-user-drag: none;
}

:host([tabindex='0']:focus-visible) {
  @include focus;
}

:host(.is-disabled) {
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.8;
}

.is-card {
  background-color: var(--card-background-color);
  color: var(--card-text-color);
  max-width: 100%;
  position: relative;

  ::slotted([slot='header']) {
    border-radius: $card-header-radius-top $card-header-radius-top $card-header-radius-bottom $card-header-radius-bottom;
    display: flex;
    align-items: stretch;
    background-color: transparent;
    padding: $card-header-padding-vertical $card-header-padding-horizontal;
  }

  ::slotted([slot='title']) {
    align-items: center;
    color: inherit;
    display: flex;
    font-size: large;
    flex-grow: 1;
    font-weight: bold;
  }

  ::slotted([slot='body']) {
    border-radius: $card-radius;
    display: block;
    background-color: transparent;
    padding: var(--card-body-padding-vertical, $card-body-padding-vertical) var(--card-body-padding-horizontal, $card-body-padding-horizontal);
  }

  ::slotted([slot='image']) {
    margin: 1rem;
    border-radius: $card-image-radius-top $card-image-radius-top $card-image-radius-bottom $card-image-radius-bottom;
  }
}
