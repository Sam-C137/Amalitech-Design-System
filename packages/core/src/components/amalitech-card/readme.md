# amalitech-card



<!-- Auto Generated Below -->


## Properties

| Property             | Attribute              | Description                                                                                   | Type                  | Default     |
| -------------------- | ---------------------- | --------------------------------------------------------------------------------------------- | --------------------- | ----------- |
| `disabled`           | `disabled`             | The disabled state of the card. If set to true, interactions would not be alowed,             | `boolean`             | `false`     |
| `href`               | `href`                 | The href attribute of the card. If set, the card would render as an anchor tag.               | `string \| undefined` | `undefined` |
| `target`             | `target`               | The target attribute of the card. If href is set, the target defaults to _blank               | `string`              | `undefined` |
| `withHoverAnimation` | `with-hover-animation` | The hover animation state of the card. If set to true, the card would have a hover animation. | `boolean`             | `false`     |


## CSS Custom Properties

| Name                             | Description                                                                                                                |
| -------------------------------- | -------------------------------------------------------------------------------------------------------------------------- |
| `--backgroundColor`              | Background color of the card, default is `#f6f6f8`;                                                                        |
| `--card-body-padding-horizontal` | Padding left and right of the card body, default is `1.25rem`;                                                             |
| `--card-body-padding-vertical`   | Padding top and bottom of the card body, default is `1.25rem`;                                                             |
| `--card-radius`                  | Border radius of the card, default is `0.5rem`;                                                                            |
| `--card-shadow`                  | Box shadow of the card, default is `0px 8px 32px 0px hsla(0, 0%, 0%, 0.1), 0px 4px 8px 0px hsla(0, 0%, 0%, 0.1)`;          |
| `--card-shadow-hover`            | Box shadow of the card on hover, default is `0px 8px 32px 0px hsla(0, 0%, 0%, 0.2), 0px 4px 8px 0px hsla(0, 0%, 0%, 0.2)`; |
| `--textColor`                    | Text color of the card, default is `#474d66`;                                                                              |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
