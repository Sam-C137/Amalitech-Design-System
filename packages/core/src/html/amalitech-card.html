<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>amalitech-card Preview</title>
    <link rel="preconnect" href="https://fonts.googleapis.com" />
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="" />
    <link
      href="https://fonts.googleapis.com/css2?family=Work+Sans:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&amp;display=swap"
      rel="stylesheet"
    />
    <script type="module" src="/build/amalitech-design-system.esm.js"></script>
    <script nomodule src="/build/build/amalitech-design-system.js"></script>
    <link rel="stylesheet" href="/build/amalitech-design-system.css" />
    <script src="https://code.iconify.design/2/2.1.0/iconify.min.js"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
    <style>
      amalitech-card {
        width: 33%;
        color: rgb(17, 19, 18);
        background-color: rgb(72, 158, 197);
        padding: 0px;

        & > div[slot='title'] {
          margin-left: 20px;
          margin-top: 10px;
        }
      }

      div[slot='body'] > div:first-of-type {
        margin: 20px 0px 0px 0px;
      }
    </style>
  </head>
  <body>
    <amalitech-card with-hover-animation href="/">
      <div slot="header">This is the header</div>
      <div slot="title">Title</div>
      <div slot="body">
        <div>Street Fighter</div>
        <div>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Accusantium animi aut, beatae dolorum fuga ipsum iusto nemo, nostrum placeat quo reiciendis reprehenderit sed
          similique soluta tenetur ut veniam veritatis, vitae?
        </div>
      </div>
      <div slot="image">
        <img src="https://th.bing.com/th/id/OIP.ITJm4XqvDqr2scT1DvvbswAAAA?rs=1&pid=ImgDetMain" alt="" />
      </div>
    </amalitech-card>
  </body>
</html>
