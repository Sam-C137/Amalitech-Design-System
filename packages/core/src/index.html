<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Stencil Component Starter</title>
    <script type="module" src="/build/amalitech-design-system.esm.js"></script>
    <script nomodule src="/build/amalitech-design-system.js"></script>
    <link rel="stylesheet" href="/build/amalitech-design-system.css" />
    <style>
      h1 {
        text-align: center;
        font-style: normal;
        line-height: 2.5rem;
        font-family: var(--amalitech-font-variant-body-family);
        color: var(--amalitech-color-neutral-700);
        font-size: var(--amalitech-font-variant-heading-3-size);
        font-weight: var(--amalitech-font-weight-bold);
        margin: 0;
        padding: 0;
      }
    </style>
  </head>

  <body>
    <h1>This is ADS</h1>
  </body>
</html>
