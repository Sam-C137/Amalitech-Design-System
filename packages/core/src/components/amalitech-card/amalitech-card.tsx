import { Component, Host, h, Prop } from '@stencil/core';
@Component({
  tag: 'amalitech-card',
  styleUrl: 'amalitech-card.scss',
  shadow: true,
})
export class AmalitechCard {
  /**
   * The disabled state of the card. If set to true, interactions would not be alowed,
   */
  @Prop() disabled: boolean = false;

  /**
   * The href attribute of the card. If set, the card would render as an anchor tag.
   */
  @Prop() href?: string;

  /**
   * The target attribute of the card. If href is set, the target defaults to _blank
   */
  @Prop() target: string;
  private clickable(): boolean {
    return this.href !== undefined;
  }
  /**
   * The hover animation state of the card. If set to true, the card would have a hover animation.
   */
  @Prop() withHoverAnimation: boolean = false;

  render() {
    const { href, target } = this;
    const TagChoose = this.clickable() ? 'a' : 'div';

    const attributes = TagChoose === 'a' ? { href, target } : {};

    return (
      <Host
        class={{
          'is-disabled': this.disabled,
          'with-hover-animation': this.withHoverAnimation,
        }}
        tabIndex={TagChoose === 'a' && !this.disabled ? 0 : -1}
      >           
        <TagChoose
          class={{
            'is-card': true,
          }}
          {...attributes}
        >
          <slot name="header"></slot>
          <slot name="title"></slot>
          <slot name="body"></slot>
          <slot name="image"></slot>
          <slot></slot>
        </TagChoose>
      </Host>
    );
  }
}
