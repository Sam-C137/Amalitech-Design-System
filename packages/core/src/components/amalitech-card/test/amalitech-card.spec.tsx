import { newSpecPage } from '@stencil/core/testing';
import { AmalitechCard } from '../amalitech-card';

jest.setTimeout(40000); // added to prevent timeout error

describe('amalitech-card', () => {
  it('renders', async () => {
    await newSpecPage({
      components: [AmalitechCard],
      html: `<amalitech-card></amalitech-card>`,
    });
  });

  it('renders a card without href', async () => {
    const page = await newSpecPage({
      components: [AmalitechCard],
      html: `<amalitech-card></amalitech-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders a clickable card with href', async () => {
    const page = await newSpecPage({
      components: [AmalitechCard],
      html: `<amalitech-card href="https://example.com"></amalitech-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders a disabled card', async () => {
    const page = await newSpecPage({
      components: [AmalitechCard],
      html: `<amalitech-card disabled></amalitech-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders card with header slot', async () => {
    const page = await newSpecPage({
      components: [AmalitechCard],
      html: `<amalitech-card><div slot="header">Header Content</div></amalitech-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
