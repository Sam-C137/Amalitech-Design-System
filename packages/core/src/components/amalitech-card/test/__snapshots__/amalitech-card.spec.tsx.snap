// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amalitech-card renders a card without href 1`] = `
<amalitech-card tabindex="-1">
  <mock:shadow-root>
    <div class="is-card">
      <slot name="header"></slot>
      <slot name="title"></slot>
      <slot name="body"></slot>
      <slot name="image"></slot>
      <slot></slot>
    </div>
  </mock:shadow-root>
</amalitech-card>
`;

exports[`amalitech-card renders a clickable card with href 1`] = `
<amalitech-card href="https://example.com" tabindex="0">
  <mock:shadow-root>
    <a class="is-card" href="https://example.com">
      <slot name="header"></slot>
      <slot name="title"></slot>
      <slot name="body"></slot>
      <slot name="image"></slot>
      <slot></slot>
    </a>
  </mock:shadow-root>
</amalitech-card>
`;

exports[`amalitech-card renders a disabled card 1`] = `
<amalitech-card class="is-disabled" disabled="" tabindex="-1">
  <mock:shadow-root>
    <div class="is-card">
      <slot name="header"></slot>
      <slot name="title"></slot>
      <slot name="body"></slot>
      <slot name="image"></slot>
      <slot></slot>
    </div>
  </mock:shadow-root>
</amalitech-card>
`;

exports[`amalitech-card renders card with header slot 1`] = `
<amalitech-card tabindex="-1">
  <mock:shadow-root>
    <div class="is-card">
      <slot name="header"></slot>
      <slot name="title"></slot>
      <slot name="body"></slot>
      <slot name="image"></slot>
      <slot></slot>
    </div>
  </mock:shadow-root>
  <div slot="header">
    Header Content
  </div>
</amalitech-card>
`;
